### ISSUE TITLE
HttpClient created by WebApplicationFactory doesn't follow HTTP status code 303 (See other) redirect

### ISSUE THREAD

**Issue by ambrosmiroslav** [dotnet/aspnetcore#33218]
The HttpClient created "manually" is able to follow HTTP status code 303 (See other) and jump to redirected page:

            _client = new HttpClient();

But the HttpClient retrieved from WebApplicationFactory isn't:

            public ClientContext(WebApplicationFactory<MyProject.Startup> Factory)
            {
                        _client = Factory.CreateClient();
            }

No matter the WebApplicationFactoryClientOptions are specified or not...

Could it be this HTTP status code is not mentioned in https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs, see the function IsRedirect()?

```
        private bool IsRedirect(HttpResponseMessage response) =>
            response.StatusCode == HttpStatusCode.MovedPermanently ||
                response.StatusCode == HttpStatusCode.Redirect ||
                response.StatusCode == HttpStatusCode.RedirectKeepVerb ||
                (int)response.StatusCode == 308;
```

**Comment by javiercn** [comment]
@ambrosmiroslav thanks for contacting us.

We would be happy to take a PR for this including the change and a test to verify the change if you want to contribute it.

**Comment by khellang** [comment]
Closed by #33246?
